source/Relativity.Testing.Framework.Api/DataCollection.cs

```
@@ -0,0 +1,25 @@
+﻿namespace Relativity.Testing.Framework.Api
+{
+	/// <summary>
+	/// Represents the Application Insights data collection state of the testing framework.
+	/// </summary>
+	public enum DataCollection
+	{
+		/// <summary>
+		/// No data is collected by the framework.
+		/// </summary>
+		None,
+
+		/// <summary>
+		/// Only non-identifying information is collected.

```

We capture Hostname on `UsageOnly` and that's potentially identifying information.

IMO Hostname should only be captured on `All` but I don't have terribly strong feelings about this.

---

I would agree with this.
If we remove that data-point though, we lose the ability to try to distinguish between internal and external users.
We should talk to Parimal about this before making a change, because I believe he will be invested in the decision.

---

- [ ] Ask Parimal about this

---

Will bring it up with him. How would you guys feel about flipping the idea a little bit? Would we be able to "tell" someone is external if we make the assumption that, if a user is internal, they will be using "All" and networking with a well-known URI/domain.

I don't think that would address internal devs using it locally without anything set.. so maybe the line is "InPipelines/Ad-Hoc" instead of "internal/external"? Just spitballing here, it seems like there's an AC catch-22 between "usage-only" and "knowing you are an external person using this".

---

Related, but different question, what about people using RTF in a black-site setup? I assume we get no insights whatsoever and that's fine/expected here?

---

It's probably just going to silently fail to report anything in that scenario.
There's nothing that we can do about that though 🤷.

---

TestVMs aren't on a domain though, but I believe the `P-DV-VM-` prefix is strictly enforced.

---

Parimal said the imperative is getting internal/external usage insight. He was expecting an opt-out collection program to be documented etc. If they actively opt-out, not getting that info is fine (but we should at least record that they opted-out)

I'll update this PR to reflect that (by defaulting `All`).